assets/asset_resolver.h

```diff
@@ -62,7 +62,8 @@ class AssetResolver {
   // Same as GetAsMapping() but returns mappings for all files who's name
   // matches |pattern|. Returns empty vector if no matching assets are found
   [[nodiscard]] virtual std::vector<std::unique_ptr<fml::Mapping>>
```

Comment:
Explain what the subdir parameter does here

Reply:
Can we make the parameter an `std::optional<std::string>` instead of an empty string please? And also document the parameters using the headerdoc format like in the rest of this header.

Reply:
done